lib/ansible/modules/cloud/amazon/elb_target.py

```diff
@@ -0,0 +1,302 @@
+    target_group_arn = module.params.get("target_group_arn")
+    target_id = module.params.get("target_id")
+    target_port = module.params.get("target_port")
+    target_status = module.params.get("target_status")
+    target_status_timeout = module.params.get("target_status_timeout")
+    changed = False
+
+    if not target_group_arn:
+        target_group_arn = convert_tg_name_to_arn(connection, module, module.params.get("target_group_name"))
+
+    target = dict(Id=target_id)
+    if target_az:
+        target['AvailabilityZone'] = target_az
+    if target_port:
+        target['Port'] = target_port
+
+    try:
+        target_description = describe_targets(connection, module, target_group_arn, [target])
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    if 'Reason' in target_description['TargetHealth']:
+        if target_description['TargetHealth']['Reason'] == "Target.NotRegistered":
+            try:
+                connection.register_targets(TargetGroupArn=target_group_arn, Targets=[target])
+                changed = True
+                if target_status:
+                    target_status_check(connection, module, target_group_arn, target, target_status, target_status_timeout)
+            except ClientError as e:
+                module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    # Get all targets for the target group
+    target_descriptions = describe_targets(connection, module, target_group_arn, [])
+
+    module.exit_json(changed=changed, target_health_descriptions=camel_dict_to_snake_dict(target_descriptions), target_group_arn=target_group_arn)
+
+
+def deregister_target(connection, module):
+
+    """
+    Deregisters a target to a target group
+
+    :param module: ansible module object
+    :param connection: boto3 connection
+    :return:
+    """
+
+    deregister_unused = module.params.get("deregister_unused")
+    target_group_arn = module.params.get("target_group_arn")
+    target_id = module.params.get("target_id")
+    target_port = module.params.get("target_port")
+    target_status = module.params.get("target_status")
+    target_status_timeout = module.params.get("target_status_timeout")
+    changed = False
+
+    if not target_group_arn:
+        target_group_arn = convert_tg_name_to_arn(connection, module, module.params.get("target_group_name"))
+
+    target = dict(Id=target_id)
+    if target_port:
+        target['Port'] = target_port
+
+    try:
+        target_description = describe_targets(connection, module, target_group_arn, [target])
+        current_target_state = target_description['TargetHealth']['State']
+    except ClientError as e:
+        module.fail_json(msg=e.message, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+
+    if current_target_state == 'unused':
```

Comment:
This module only deregisters a module if `current_target_state` is `unused`. I would like to use it to deregister used targets. This fails silently now.

Reply:
Yes, from what I can see this doesn't work now. My `state: absent` is ignored when target is used by target group. Where is `else`? :)

Reply:
yeah, there's a missing chunk of code.